### PR TITLE
Provide consistent examples of pub-sub for all SDKs

### PR DESCRIPTION
**Pull request by amulyavarote**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**

- [ ] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Modified pub sub documentation page (under building blocks) for dapr.io

## Issue reference

This partially fulfills: #1969

### CODE REVIEW

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md [hunk 1]
```diff
@@ -20,33 +20,48 @@ When publishing a message, it's important to specify the content type of the dat
 Unless specified, Dapr will assume `text/plain`. When using Dapr's HTTP API, the content type can be set in a `Content-Type` header.
 gRPC clients and SDKs have a dedicated content type parameter.
 
-## Step 1: Setup the Pub/Sub component
-The following example creates applications to publish and subscribe to a topic called `deathStarStatus`.
+## Example:
+
+The below code examples loosely describes an application that processes orders. In the examples, there are two services - an order processing service and a checkout service. Both services have Dapr sidecars. The order processing service uses Dapr to publish message and the checkout service subscribes to the message in Rabbit mq.
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md]:
```suggestion
The below code examples loosely describes an application that processes orders. In the examples, there are two services - an order processing service and a checkout service. Both services have Dapr sidecars. The order processing service uses Dapr to publish a message and the checkout service subscribes to the message in Rabbit mq.
```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md [hunk 2]
```diff
@@ -20,33 +20,48 @@ When publishing a message, it's important to specify the content type of the dat
...
 
 The first step is to setup the Pub/Sub component:
 
 {{< tabs "Self-Hosted (CLI)" Kubernetes >}}
 
 {{% codetab %}}
-Redis Streams is installed by default on a local machine when running `dapr init`.
+pubsub.yaml is created by default on a local machine when running `dapr init`. Verify by opening your components file under `%UserProfile%\.dapr\components\pubsub.yaml` on Windows or `~/.dapr/components/pubsub.yaml` on Linux/MacOS.
+
+In this example, rabbit mq is used for publish and subscribe. Replace pubsub.yaml file contents with the below contents.
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md]:
```suggestion
In this example, RabbitMQ is used for publish and subscribe. Replace pubsub.yaml file contents with the below contents.
```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md [hunk 3]
```diff
@@ -20,33 +20,48 @@ When publishing a message, it's important to specify the content type of the dat
 Unless specified, Dapr will assume `text/plain`. When using Dapr's HTTP API, the content type can be set in a `Content-Type` header.
 gRPC clients and SDKs have a dedicated content type parameter.
 
-## Step 1: Setup the Pub/Sub component
-The following example creates applications to publish and subscribe to a topic called `deathStarStatus`.
+## Example:
+
+The below code examples loosely describes an application that processes orders. In the examples, there are two services - an order processing service and a checkout service. Both services have Dapr sidecars. The order processing service uses Dapr to publish a message and the checkout service subscribes to the message in RabbitMQ.
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md]:
```suggestion
The below code example loosely describes an application that processes orders. In the example, there are two services - an order processing service and a checkout service. Both services have Dapr sidecars. The order processing service uses Dapr to publish a message to RabbitMQ and the checkout service subscribes to the topic in the message queue.
```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md [hunk 4]
```diff
@@ -20,33 +20,48 @@ When publishing a message, it's important to specify the content type of the dat
...
+## Step 1: Setup the Pub/Sub component
+The following example creates applications to publish and subscribe to a topic called `orders`.
 
 The first step is to setup the Pub/Sub component:
 
 {{< tabs "Self-Hosted (CLI)" Kubernetes >}}
 
 {{% codetab %}}
-Redis Streams is installed by default on a local machine when running `dapr init`.
+pubsub.yaml is created by default on a local machine when running `dapr init`. Verify by opening your components file under `%UserProfile%\.dapr\components\pubsub.yaml` on Windows or `~/.dapr/components/pubsub.yaml` on Linux/MacOS.
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md]:
```suggestion
The pubsub.yaml is created by default on your local machine when running `dapr init`. Verify by opening your components file under `%UserProfile%\.dapr\components\pubsub.yaml` on Windows or `~/.dapr/components/pubsub.yaml` on Linux/MacOS.
```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md [hunk 5]
```diff
@@ -20,33 +20,48 @@ When publishing a message, it's important to specify the content type of the dat
...
 
-## Step 1: Setup the Pub/Sub component
-The following example creates applications to publish and subscribe to a topic called `deathStarStatus`.
+## Example:
+
+The below code example loosely describes an application that processes orders. In the example, there are two services - an order processing service and a checkout service. Both services have Dapr sidecars. The order processing service uses Dapr to publish a message to RabbitMQ and the checkout service subscribes to the topic in the message queue.
 
-<img src="/images/pubsub-publish-subscribe-example.png" width=1000>
-<br></br>
+<img src="/images/building-block-pub-sub-example.png" width=1000 alt="Diagram showing state management of example service">
```
Comment (willtsai) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md:27]: @greenie-msft pointed this out to me when we viewed the rendered page - there is a grey line on the right hand side of the picture that needs to be cropped out.

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md [hunk 6]
```diff
@@ -20,33 +20,48 @@ When publishing a message, it's important to specify the content type of the dat
...
 
 The first step is to setup the Pub/Sub component:
 
 {{< tabs "Self-Hosted (CLI)" Kubernetes >}}
 
 {{% codetab %}}
-Redis Streams is installed by default on a local machine when running `dapr init`.
+The pubsub.yaml is created by default on your local machine when running `dapr init`. Verify by opening your components file under `%UserProfile%\.dapr\components\pubsub.yaml` on Windows or `~/.dapr/components/pubsub.yaml` on Linux/MacOS.
+
+In this example, RabbitMQ is used for publish and subscribe. Replace pubsub.yaml file contents with the below contents.
```
Comment (willtsai) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md]: nitpick:
```suggestion
In this example, RabbitMQ is used for publish and subscribe. Replace `pubsub.yaml` file contents with the below contents.
```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md [hunk 7]
```diff
@@ -95,390 +121,508 @@ You can subscribe to a topic using the following Custom Resources Definition (CR
...
+dapr run --app-id orderprocessing --app-port 6001 --dapr-http-port 3601 --dapr-grpc-port 60001 npm start
 ```
 
 {{% /codetab %}}
 
 {{< /tabs >}}
 
+## Step 4: ACK-ing a message
+
+In order to tell Dapr that a message was processed successfully, return a `200 OK` response. If Dapr receives any other return status code than `200`, or if your app crashes, Dapr will attempt to redeliver the message following At-Least-Once semantics.
```
Comment (willtsai) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md]:
```suggestion
In order to tell Dapr that a message was processed successfully, return a `200 OK` response. If Dapr receives any other return status code than `200`, or if your app crashes, Dapr will attempt to redeliver the message following at-least-once semantics.
```

**Review by willtsai — CHANGES_REQUESTED** [review]

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md (hunk 5 above)
Comment (amulyavarote) [reply daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md:27]: I removed it.

**Review by willtsai — APPROVED** [review]

**Review by greenie-msft — APPROVED** [review]

**Pull request merged**